apix-impl/src/main/java/eu/xenit/apix/alfresco/metadata/NodeService.java

```diff
@@ -421,6 +426,56 @@ public NodeAssociations getAssociations(eu.xenit.apix.data.NodeRef ref) {
         return new NodeAssociations(getChildAssociations(ref), getParentAssociations(ref), getTargetAssociations(ref));
     }
 
+    @Override
+    public List<eu.xenit.apix.data.NodeRef> getRecursiveParents(eu.xenit.apix.data.NodeRef ref, eu.xenit.apix.data.NodeRef rootRef) {
+        NodeRef alfrescoRootRef;
+        if (rootRef == null){
+            alfrescoRootRef = repository.getCompanyHome();
+        }
+        else
+        {
+            alfrescoRootRef = c.alfresco(rootRef);
+        }
+
+        NodeRef nodeRef = c.alfresco(ref);
+        if (nodeRef.equals(alfrescoRootRef)){
+            return new ArrayList<>();
+        }
+
+        List<eu.xenit.apix.data.NodeRef> recursiveParentRefs = new ArrayList<>();
+        if (!nodeService.exists(nodeRef)) return null;
+        if (permissionService.hasReadPermission(nodeRef) != AccessStatus.ALLOWED) return null;
+
+        ChildAssociationRef childAssocRef = nodeService.getPrimaryParent(nodeRef);
+        if (childAssocRef == null) return null;
+
+        NodeRef parentRef = childAssocRef.getParentRef();
+        while (parentRef != null) {
+            recursiveParentRefs.add(c.apix(parentRef));
+            if (parentRef.equals(alfrescoRootRef)) break;
+            if (permissionService.hasReadPermission(parentRef) != AccessStatus.ALLOWED) return null;
+            ChildAssociationRef parentAssoc = nodeService.getPrimaryParent(parentRef);
+            if (parentAssoc == null) return null;
+            parentRef = parentAssoc.getParentRef();
+        }
+
+        return recursiveParentRefs;
+    }
+
+    @Override
+    public Map<eu.xenit.apix.data.NodeRef, List<eu.xenit.apix.data.NodeRef>> getRecursiveParents(
```

Comment:
Since I give the suggestion to go for single-object calls, this function can be removed.